proto/metrics_test.go

```diff
@@ -32,30 +32,32 @@ func (s *MetricsTestSuite) SetupTest() {
 }
 
 func (s *MetricsTestSuite) TestRunForIntervalAndSendMessages() {
+	s.comm.Mu.Lock()
 	s.Zero(len(s.comm.ProcInfo[s.id]))
```

Comment:
Would it make more sense to create a getter in the mock client that does the locking and returns ProcInfo? It will make this code a lot cleaner